crates/aptos/src/node/local_testnet/mod.rs

```diff
@@ -0,0 +1,426 @@
+};
+use tokio::task::JoinHandle;
+use tracing_subscriber::fmt::MakeWriter;
+
+const TESTNET_FOLDER: &str = "testnet";
+
+/// Run a local testnet
+///
+/// This local testnet will run it's own genesis and run as a single node network
+/// locally. A faucet and grpc transaction stream will run alongside the node unless
+/// you specify otherwise with --no-faucet and --no-txn-stream respectively.
+#[derive(Parser)]
+pub struct RunLocalTestnet {
+    /// An overridable config template for the test node
+    ///
+    /// If provided, the config will be used, and any needed configuration for the local testnet
+    /// will override the config's values
+    #[clap(long, value_parser)]
+    config_path: Option<PathBuf>,
+
+    /// The directory to save all files for the node
+    ///
+    /// Defaults to .aptos/testnet
+    #[clap(long, value_parser)]
+    test_dir: Option<PathBuf>,
+
+    /// Path to node configuration file override for local test mode.
+    ///
+    /// If provided, the default node config will be overridden by the config in the given file.
+    /// Cannot be used with --config-path
+    #[clap(long, value_parser, conflicts_with("config_path"))]
+    test_config_override: Option<PathBuf>,
+
+    /// Random seed for key generation in test mode
+    ///
+    /// This allows you to have deterministic keys for testing
+    #[clap(long, value_parser = aptos_node::load_seed)]
+    seed: Option<[u8; 32]>,
+
+    /// Clean the state and start with a new chain at genesis
+    ///
+    /// This will wipe the aptosdb in `test-dir` to remove any incompatible changes, and start
+    /// the chain fresh.  Note, that you will need to publish the module again and distribute funds
+    /// from the faucet accordingly
+    #[clap(long)]
+    force_restart: bool,
+
+    /// Port to run the faucet on.
+    ///
+    /// When running, you'll be able to use the faucet at `http://127.0.0.1:<port>/mint` e.g.
+    /// `http//127.0.0.1:8081/mint`
+    #[clap(long, default_value_t = 8081)]
+    faucet_port: u16,
+
+    /// Do not run a faucet alongside the node.
+    ///
+    /// Running a faucet alongside the node allows you to create and fund accounts
+    /// for testing.
+    #[clap(long)]
+    no_faucet: bool,
+
+    /// This does nothing, we already run a faucet by default. We only keep this here
+    /// for backwards compatibility with tests. We will remove this once the commit
+    /// that added --no-faucet makes its way to the testnet branch.
+    #[clap(long, hide = true)]
+    with_faucet: bool,
+
+    /// Disable the delegation of faucet minting to a dedicated account.
+    #[clap(long)]
+    do_not_delegate: bool,
+
+    /// Do not run a transaction stream service alongside the node.
+    ///
+    /// Note: In reality this is not the same as running a Transaction Stream Service,
+    /// it is just using the stream from the node, but in practice this distinction
+    /// shouldn't matter.
+    #[clap(long)]
+    no_txn_stream: bool,
```

Comment:
How much overhead does this create?

Mostly concerned since we've had to tune the local testnet several times, it can be a lot of overhead on local machines.

Reply:
I ran some tests and I found the CPU overhead to be negligible. If you don't connect to the stream then this doesn't really do anything. I think maybe it enables some extra tracking in storage for tables (which will be moved out as some point).